packages/controls/src/phosphor/tabpanel.ts

```diff
@@ -130,7 +130,7 @@ class TabPanel extends Widget {
    * Set the index of the currently selected tab.
    *
    * #### Notes
-   * If the index is out of range, it will be set to `-1`.
+   * If the index is out of range, it will be set to `null`.
```

Comment:
Actually, you don't need to do anything here - the tab bar handles the logic to set its current index to -1 if out of range. We'll just handle the translation to null up where we retrieve the index.